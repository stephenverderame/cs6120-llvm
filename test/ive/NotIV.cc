// NOLINTBEGIN
#include <cstdio>
int main()
{
    int m = 20;
    for (int i = 0; i < 5000; i += m) {
        int k = i * i;
        int j = m * m;
        int f = 10 - i;
        printf("%d\n", k + j + f);
    }
}
// NOLINTEND