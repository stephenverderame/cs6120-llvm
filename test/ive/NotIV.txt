410
790
1970
3950
6730
10310
14690
19870
25850
32630
40210
48590
57770
67750
78530
90110
102490
115670
129650
144430
160010
176390
193570
211550
230330
249910
270290
291470
313450
336230
359810
384190
409370
435350
462130
489710
518090
547270
577250
608030
639610
671990
705170
739150
773930
809510
845890
883070
921050
959830
999410
1039790
1080970
1122950
1165730
1209310
1253690
1298870
1344850
1391630
1439210
1487590
1536770
1586750
1637530
1689110
1741490
1794670
1848650
1903430
1959010
2015390
2072570
2130550
2189330
2248910
2309290
2370470
2432450
2495230
2558810
2623190
2688370
2754350
2821130
2888710
2957090
3026270
3096250
3167030
3238610
3310990
3384170
3458150
3532930
3608510
3684890
3762070
3840050
3918830
3998410
4078790
4159970
4241950
4324730
4408310
4492690
4577870
4663850
4750630
4838210
4926590
5015770
5105750
5196530
5288110
5380490
5473670
5567650
5662430
5758010
5854390
5951570
6049550
6148330
6247910
6348290
6449470
6551450
6654230
6757810
6862190
6967370
7073350
7180130
7287710
7396090
7505270
7615250
7726030
7837610
7949990
8063170
8177150
8291930
8407510
8523890
8641070
8759050
8877830
8997410
9117790
9238970
9360950
9483730
9607310
9731690
9856870
9982850
10109630
10237210
10365590
10494770
10624750
10755530
10887110
11019490
11152670
11286650
11421430
11557010
11693390
11830570
11968550
12107330
12246910
12387290
12528470
12670450
12813230
12956810
13101190
13246370
13392350
13539130
13686710
13835090
13984270
14134250
14285030
14436610
14588990
14742170
14896150
15050930
15206510
15362890
15520070
15678050
15836830
15996410
16156790
16317970
16479950
16642730
16806310
16970690
17135870
17301850
17468630
17636210
17804590
17973770
18143750
18314530
18486110
18658490
18831670
19005650
19180430
19356010
19532390
19709570
19887550
20066330
20245910
20426290
20607470
20789450
20972230
21155810
21340190
21525370
21711350
21898130
22085710
22274090
22463270
22653250
22844030
23035610
23227990
23421170
23615150
23809930
24005510
24201890
24399070
24597050
24795830
