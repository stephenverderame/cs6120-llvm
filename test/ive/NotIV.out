Loop: 
1:                                                ; preds = %11, %0
  %.0 = phi i32 [ 0, %0 ], [ %12, %11 ]
  %2 = icmp slt i32 %.0, 5000
  br i1 %2, label %4, label %3

Basic IV:   %.0 = phi i32 [ 0, %0 ], [ %12, %11 ]
Basic IV:   %12 = add nsw i32 %.0, 20
